### PR TITLE
refactor: reimplement navigator tree with new component

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio/issues/3399

What to test

- hover
- select
- hiding (with show button) item should be transparent
- slot should be purple
- hide slot fragment
- render collections
- enter should trigger edit instance text
- support editing instance label with cmd+e
- should prevent dragging rich text child
- should prevent dragging non detachable instances (accordion header)
- should support reparenting after the last child

### CODE REVIEW

File: packages/design-system/src/components/tree.tsx [hunk 1]
```diff
@@ -215,24 +215,45 @@ export type TreeDropTarget = {
...
+    };
+  }
+  if (treeDropTarget?.afterLevel !== undefined) {
+    return {
+      type: "reorder-below",
+      currentLevel: treeDropTarget.afterLevel,
+      indentPerLevel: BARS_GAP,
+    };
+  }
+  return null;
```
Comment (kof) [inline packages/design-system/src/components/tree.tsx:256]: is this old? should be using undefined, basically no return

File: packages/design-system/src/components/list-position-indicator.tsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ const OVERLAP = 2;
 
 const Container = styled(Box, {
   position: "absolute",
+  zIndex: 1,
```
Comment (kof) [inline packages/design-system/src/components/list-position-indicator.tsx:19]: would isolation: isolate work here?

Comment (TrySound) [reply packages/design-system/src/components/list-position-indicator.tsx:19]: I don't think so, I just need to get it higher than outline

**Review by johnsicili — CHANGES_REQUESTED** [review]
Works really well!

1. Adding new nested items should be expanded by default (add box, add heading, can't see heading because box is collapsed). @kof fixed this IIRC when he was doing shortcuts and search for components.
2. Related to the first, if you click on an item that is nested, you can't see anything selected in the navigator. Same with hovering
3. When there are enough instances in the navigator to fill the viewport, and you add a new component, the new one is not scrolled into view in the navigator

**Comment by TrySound** [comment]
@johnsicili
fixed 1 and 2 though, we never expanded on hover, should we?
Scrolling works for me, could you record a video pls?

**Comment by johnsicili** [comment]
Didn't realize the hover was existing behavior. Idk what's right here so we can leave it.

Here is the no scroll... seems to happen on page load and if I select the parent via the canvas

https://github.com/user-attachments/assets/1b749bf2-6417-4419-8c2d-884e5719b19e

**Review by kof — APPROVED** [review]

**Comment by TrySound** [comment]
Tree scroll seems like fixed. Though found we have a weird problem with auto scroll on canvas. Reproduced in main as well so not a blocker for this PR.

**Review by johnsicili — APPROVED** [review]
Nice! Honestly I think I like it better without smooth scroll. It's faster

**Pull request merged**